specs/gluster_cluster_list_view.adoc

```diff
@@ -0,0 +1,140 @@
+// vim: tw=79
+
+= Tendrl Definitions: Display list of clusters discovered by Tendrl
+
+Display list of Gluster clusters present in Tendrl. The list should have functionality for 
+
+* filtering by name.
+* sorting by name, status, cluster version and managed.
+
+Cluster list will be sorted by name by default.
+
+== Problem description
+
+Tendrl should show list of Gluster clusters present by Tendrl along with their details.
+
+The details shown for each cluster in UI is -
+
+* Cluster Name
+* Status
+* Managed State
+* Cluster Version
+* Import button(if ready to import)
+* Launch Dashboard button(if already imported)
+* Cluster Details button(if already imported)
+* Message to state cluster configurable state for eg. ready to import, cluster misconfigured.
+
+The workflow is as follows:
+
+* API will provide the list of clusters to the Tendrl.
+
+* User can click on expand icon of each cluster record to see the error messages/logs and hosts associated with it.
+
+* Based on the managed state, import/launch dasboard/cluster details buttons will be displayed for each cluster.
+  * If Managed state is "Yes", Launch dashboard and cluster details button will be displayed.
+  * If Managed state is "No" and cluster is ready to import, Import button will be displayed.
+  * If Managed state is "No" and cluster is not properly configured, Imported will be disabled.
+
+* Clicking Import button will take to import cluster workflow.
+
+* Clicking Launch dashboard button will launch the dashboard in a new window.
+
+* Clicking Cluster details button will take to cluster objects details page.
+
+== Use Casess
+
+Users can see the list of cluster present in Tendrl. They can also 
+
+* filtering by name.
+* sorting by name, status, cluster version and managed.
+
+== Proposed change
+
+* Create a view for listing clusters.
+* Use the patternfly expanded list views design to display list of clusters.
+* Implement client side filer and sorting functionality using angular's filter and orderby components.
+* Send polling call after 10 seconds to get the updated data. 
+* Show total count of clusters present in Tendrl.
+
+Tendrl API related changes:
+
+* Get API call for getting cluster list.
+
+=== Alternatives
+
+Generate list component can be created and used for listing the cluster. But because of its complexity and time constraint, we will pick this up later.
+
+=== Data model impact:
+See 'Impacted Modules' section where all the details are populated.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+/api/1.0/GetClusterList - fetch the list of clusters
+
+Sample API Response - required.
```

Comment:
@anivargi @brainfunked @nthomas-redhat As the cluster list will list all the clusters which are already imported and ready to import, I think it make sense to have a single API call for it. As per the current API, cluster which are available for importing is present inside GetNodeList API and imported cluster is in GetClusterList API.